specs/core/1_new_shards.md

```diff
@@ -0,0 +1,583 @@
+        - [`get_updated_gasprice`](#get_updated_gasprice)
+        - [`is_valid_indexed_attestation`](#is_valid_indexed_attestation)
+        - [`get_attestation_shard`](#get_attestation_shard)
+    - [Beacon Chain Changes](#beacon-chain-changes)
+        - [New beacon state fields](#new-beacon-state-fields)
+        - [New beacon block data fields](#new-beacon-block-data-fields)
+        - [Attestation processing](#attestation-processing)
+            - [`validate_attestation`](#validate_attestation)
+            - [`apply_shard_transition`](#apply_shard_transition)
+            - [`process_attestations`](#process_attestations)
+        - [Misc block post-processing](#misc-block-post-processing)
+        - [Light client processing](#light-client-processing)
+        - [Epoch transition](#epoch-transition)
+    - [Fraud proofs](#fraud-proofs)
+    - [Shard state transition function](#shard-state-transition-function)
+    - [Honest committee member behavior](#honest-committee-member-behavior)
+
+<!-- /TOC -->
+
+## Introduction
+
+This document describes the shard transition function (data layer only) and the shard fork choice rule as part of Phase 1 of Ethereum 2.0.
+
+## Configuration
+
+### Misc
+
+| Name | Value | Unit | Duration |
+| - | - | - | - | 
+| `MAX_SHARDS` | `2**10` (= 1024) |
+| `ACTIVE_SHARDS` | `2**6` (= 64) |
+| `ONLINE_PERIOD` | `2**3` (= 8) | epochs | ~51 min |
+| `LIGHT_CLIENT_COMMITTEE_SIZE` | `2**7` (= 128) |
+| `LIGHT_CLIENT_COMMITTEE_PERIOD` | `2**8` (= 256) | epochs | ~27 hours |
+| `SHARD_COMMITTEE_PERIOD` | `2**8` (= 256) | epochs | ~27 hours |
+| `SHARD_BLOCK_CHUNK_SIZE` | `2**18` (= 262,144) | |
+| `MAX_SHARD_BLOCK_CHUNKS` | `2**2` (= 4) | |
+| `BLOCK_SIZE_TARGET` | `3 * 2**16` (= 196,608) | |
+| `SHARD_BLOCK_OFFSETS` | `[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233]` | |
+| `MAX_SHARD_BLOCKS_PER_ATTESTATION` | `len(SHARD_BLOCK_OFFSETS)` | |
+| `EMPTY_CHUNK_ROOT` | `hash_tree_root(BytesN[SHARD_BLOCK_CHUNK_SIZE]())` | |
+| `MAX_GASPRICE` | `2**14` (= 16,384) | Gwei | |
+| `MIN_GASPRICE` | `2**5` (= 32) | Gwei | |
+| `GASPRICE_ADJUSTMENT_COEFFICIENT` | `2**3` (= 8) | |
+| `DOMAIN_SHARD_LIGHT_CLIENT` | `192` | |
+| `DOMAIN_SHARD_COMMITTEE` | `192` | |
+| `DOMAIN_SHARD_PROPOSAL` | `193` | |
+
+## Containers
+
+### `ShardBlockWrapper`
+
+_Wrapper for being broadcasted over the network._
+
+```python
+class ShardBlockWrapper(Container):
+    shard_parent_root: Hash
+    beacon_parent_root: Hash
+    slot: Slot
+    body: BytesN[SHARD_BLOCK_CHUNK_SIZE]
+    signature: BLSSignature
+```
+
+### `ShardSignableHeader`
+
+```python
+class ShardSignableHeader(Container):
+    shard_parent_root: Hash
+    beacon_parent_root: Hash
+    slot: Slot
+    body_root: Hash
+```
+
+### `ShardState`
+
+```python
+class ShardState(Container):
+    slot: Slot
+    gasprice: Gwei
+    data: Hash
+    latest_block_root: Hash
+```
+
+### `AttestationData`
+
+```python
+class AttestationData(Container):
```

Comment:
Would it make sense to add `shard` to the field?  can avoid `get_shard`

Reply:
Would still need to validate the shard against the committee index for the slot, but I see value in including it here explicitly (similar to including `slot` even though it used to be able to be calculated from the constituent parts)